src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -525,12 +550,38 @@ public SCMFileSystem build(@NonNull SCMSource source, @NonNull SCMHead head,
             if (owner == null) {
                 throw new IOException("Cannot instantiate a SCMFileSystem from an SCM without an owner");
             }
-            return build(owner, source.build(head, rev), rev);
+            return build(owner, source.build(head, rev), rev, null);
         }
 
         private boolean isEnclosedByDescribable(Descriptor<?> descriptor) {
             Class<?> enclosingClass = getClass().getEnclosingClass();
             return enclosingClass != null && descriptor.clazz.isAssignableFrom(enclosingClass);
         }
+
+        /**
+         * Given a {@link SCM} this should try to build a corresponding {@link SCMFileSystem} instance that
+         * reflects the content at the specified {@link SCMRevision}. If the {@link SCM} is supported but not
+         * for a fixed revision, best effort is acceptable as the most capable {@link SCMFileSystem} will be returned
+         * to the caller. If the {@link Run} is provided, it can be used to alter the behaviour of buil based on the current Run
+         * for example, expand branch name based on current build properties, etc.
```

Comment:
There is a typo here `buil` -> `build`, but I might reword things a bit:

```suggestion
         * to the caller. If the {@link Run} is provided, it can be used to alter the behavior. 
         * For example, variables in the branch name can be expanded based on current build properties, etc.
```